test2code/src/main/kotlin/com/epam/drill/plugins/test2code/Plugin.kt

```diff
@@ -246,6 +202,22 @@ class Plugin(
         }
         logger.info { "Scanned $classCount classes" }
     }
+
+    //TODO remove after admin refactoring
+    private fun createSession(
+        sessionId: String,
+        isRealtime: Boolean = true, isGlobal: Boolean = false
+    ) {
+        if (sessions[sessionId] != null)
+            return
+        synchronized(sessionId.intern()) {
+            if (sessions[sessionId] != null)
```

Comment:
- style preference: do not place a single op on the new line (prefer `if(condition) return`)
- reasoning for the duplicate check in synchronized block

Reply:
double-check locking